shared/src/cred.rs

```diff
@@ -55,40 +203,203 @@ impl CredentialRPK {
 
             if g_a_x_offset
                 .checked_add(P256_ELEM_LEN)
-                .map_or(false, |end| end <= cred.len())
+                .map_or(false, |end| end <= value.len())
             {
-                Ok((
-                    cred[g_a_x_offset..g_a_x_offset + P256_ELEM_LEN]
-                        .try_into()
-                        .expect("Wrong key length"),
-                    cred[id_cred_offset],
-                ))
+                let public_key: BytesKeyEC2 = value[g_a_x_offset..g_a_x_offset + P256_ELEM_LEN]
+                    .try_into()
+                    .expect("Wrong key length");
+                let kid = value[id_cred_offset];
+
+                Ok(Self {
+                    bytes: BufferCred::new_from_slice(value)
+                        .map_err(|_| EDHOCError::ParsingError)?,
+                    key: CredentialKey::EC2Compact(public_key),
+                    kid: Some(BufferKid::new_from_slice(&[kid]).unwrap()),
+                    cred_type: CredentialType::CCS,
+                })
             } else {
                 Err(EDHOCError::ParsingError)
             }
         }
     }
+
```

Comment:
I believe the PSK parsing routine should be added on top of the current PR as a separate branch, which we can merge once the PSK method becomes more stable

Reply:
For now I documented it as experimental and moved the tests to a dedicated `test_experimental` module.